src/machine/machined-varlink.c

```diff
@@ -415,32 +417,78 @@ static int list_machine_one(sd_varlink *link, Machine *m, bool more) {
         return sd_varlink_reply(link, v);
 }
 
+typedef struct MachineListQueryParams {
+    const char *machine_name;
+    pid_t pid;
+} MachineListQueryParams;
+
 static int vl_method_list(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
         static const sd_json_dispatch_field dispatch_table[] = {
-                { "name", SD_JSON_VARIANT_STRING, sd_json_dispatch_const_string, 0, 0 },
+                { "name", SD_JSON_VARIANT_STRING,   sd_json_dispatch_const_string, offsetof(MachineListQueryParams, machine_name), 0 },
+                { "pid",  SD_JSON_VARIANT_UNSIGNED, sd_json_dispatch_uint32,       offsetof(MachineListQueryParams, pid),          0 },
                 {}
         };
 
-        Manager *m = ASSERT_PTR(userdata);
-        const char *mn = NULL;
         int r;
+        Manager *m = ASSERT_PTR(userdata);
+        Machine *machine = NULL, *pid_machine = NULL;
+        MachineListQueryParams p = {
+                .machine_name = NULL,
```

Comment:
Please drop `NULL` or `0` lines in structured initializer.

Reply:
this is necessary in this case. "name" is not mandatory param, so it can be omitted. Then, machine_name would be an uninitialized value.

Reply:
@ikruglov Any fields not set explicitly in an initializer such as this are initialized to `NULL`, so this can be dropped